Assets/Scripts/UI/RestartRemakeControllers/RestartRemakeController.cs

```diff
@@ -26,42 +26,47 @@
 using Maes.Simulation.SimulationScenarios;
 
 using UnityEngine;
-using UnityEngine.UI;
+using UnityEngine.UIElements;
 
-namespace Maes.UI.RestartRemakeContollers
+namespace Maes.UI.RestartRemakeControllers
 {
     public abstract class RestartRemakeController<TSimulation, TAlgorithm, TScenario> : MonoBehaviour
         where TSimulation : class, ISimulation<TSimulation, TAlgorithm, TScenario>
         where TAlgorithm : IAlgorithm
         where TScenario : SimulationScenario<TSimulation, TAlgorithm>
     {
-
-        public Button RestartCurrentButton = null!;
-        public Button RestartAllButton = null!;
-        public Button MakeAndRunButton = null!;
-        public Button CreateBatchButton = null!;
+        public UIDocument uiDocument = null!;
         public SimulationManager<TSimulation, TAlgorithm, TScenario> simulationManager = null!;
 
+        // Set by Start
+        private Button _restartCurrentButton = null!;
+        private Button _restartAllButton = null!;
+        private Button _makeAndRunButton = null!;
+        private Button _createBatchButton = null!;
+
 
-        // Start is called before the first frame update
         private void Start()
         {
-            RestartCurrentButton.onClick.AddListener(RestartCurrentScenario);
+            _restartCurrentButton = uiDocument.rootVisualElement.Q<Button>("RestartCurrentButton");
+            _restartAllButton = uiDocument.rootVisualElement.Q<Button>("RestartAllButton");
 
-            RestartAllButton.onClick.AddListener(RestartAllScenarios);
+            _makeAndRunButton = uiDocument.rootVisualElement.Q<Button>("MakeAndRunButton");
+            _createBatchButton = uiDocument.rootVisualElement.Q<Button>("CreateBatchButton");
 
-            MakeAndRunButton.onClick.AddListener(() =>
-            {
-                Debug.LogWarning("Does nothing");
-            });
+            // TODO: What is the point of this panel?
```

Comment:
Then remove them, there is no point in having them.

Reply:
This is new ui stuff